presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -793,11 +793,11 @@ protected Type visitFunctionCall(FunctionCall node, StackableAstVisitorContext<C
                                         parameters,
                                         isDescribe);
                                 if (context.getContext().isInLambda()) {
-                                    for (LambdaArgumentDeclaration argument : context.getContext().getNameToLambdaArgumentDeclarationMap().values()) {
+                                    for (LambdaArgumentDeclaration argument : context.getContext().getFieldToLambdaArgumentDeclaration().values()) {
                                         innerExpressionAnalyzer.setExpressionType(argument, getExpressionType(argument));
                                     }
                                 }
-                                return innerExpressionAnalyzer.analyze(expression, scope, context.getContext().expectingLambda(types)).getTypeSignature();
+                                return innerExpressionAnalyzer.analyze(expression, baseScope, context.getContext().expectingLambda(types)).getTypeSignature();
```

Comment:
`context.getContext().getScope()`? Definitely `expectingLambda` preserves `this.nameToLambdaArgumentDeclarationMap` in current master. Add test for this case (some kind of nested lambdas with reference to outer lambda argument)?

Reply:
I guess the problem is with this `Map<Integer, Type>` map that is used to resolve `FieldReferences` (this information needs to be passed to `innerExpressionAnalyzer`). However, https://github.com/prestodb/presto/pull/9099#discussion_r145956962 might work here. It's strange though that no test failed (lack of proper test?)

Reply:
The right `Scope` is passed via `context.getContext().expectingLambda(types)` expression, so this is correct.